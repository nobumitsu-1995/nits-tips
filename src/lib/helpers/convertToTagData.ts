import type { MicroCMS } from '@/types/microCMS'

export const convertToTagData = (
  tags: (MicroCMS['tag'] | MicroCMS['category'])[],
  isCategory: boolean = false,
) =>
  tags.map((data) => ({
    id: data.id,
    label: data.label,
    href: `/search/result/?${isCategory ? 'category' : 'tag'}=${data.id}`,
    src: 'icon' in data ? data.icon.url : '',
  }))
