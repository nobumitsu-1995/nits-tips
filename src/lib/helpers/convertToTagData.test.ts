import { originTagData } from '@/testUtils/fixtures/originTagData'
import { convertToTagData } from './convertToTagData'

test('タグ配列を表示用に整形する', () => {
  const result = convertToTagData([originTagData])
  expect(result).toEqual([
    {
      id: 'o39i0-i8y7r',
      href: '/search/result/?tag=o39i0-i8y7r',
      label: 'React',
      src: 'https://images.microcms-assets.io/assets/1ea46e2b90a147b58b1b144f170fce22/eb4b4b189c1c41c19b57e5615f1d46a6/react_icon_130845.svg',
    },
  ])
})
