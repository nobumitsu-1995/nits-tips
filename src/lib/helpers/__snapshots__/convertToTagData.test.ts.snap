// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`カテゴリ配列を表示用に整形する 1`] = `
[
  {
    "href": "/search/result/?category=daz0qkbj6",
    "id": "daz0qkbj6",
    "label": "フロントエンド",
    "src": "",
  },
  {
    "href": "/search/result/?category=zovf4yshp3",
    "id": "zovf4yshp3",
    "label": "バックエンド",
    "src": "",
  },
  {
    "href": "/search/result/?category=air585mxa",
    "id": "air585mxa",
    "label": "その他",
    "src": "",
  },
]
`;

exports[`タグ配列を表示用に整形する 1`] = `
[
  {
    "href": "/search/result/?tag=o39i0-i8y7r",
    "id": "o39i0-i8y7r",
    "label": "React",
    "src": "https://images.microcms-assets.io/assets/1ea46e2b90a147b58b1b144f170fce22/eb4b4b189c1c41c19b57e5615f1d46a6/react_icon_130845.svg",
  },
]
`;
